### PR TITLE
Hubo

### PR DESCRIPTION
**Pull request by hubo0831**
增强DatabaseTarget功能，支持参数类型和格式

----

Enhanced DatabaseTarget functionality, supporting parameter types and formats

### CODE REVIEW

**Comment by 304NotModified** [comment]
Thanks for the submit!

Could you check this?
>  Generated session id: 7e0087e4991050754d560326f0c21f30
  Generated build app name: NLog.Xamarin.iOs
Targets\DatabaseParameterInfo.cs(34,14): error CS0234: The type or namespace name 'Data' does not exist in the namespace 'System' (are you missing an assembly reference?) [C:\projects\nlog\src\NLog\NLog.Xamarin.iOs.csproj]
Targets\DatabaseParameterInfo.cs(94,18): error CS0246: The type or namespace name 'DbType' could not be found (are you missing a using directive or an assembly reference?) [C:\projects\nlog\src\NLog\NLog.Xamarin.iOs.csproj]

Also, could you add some unit tests? There are already unit tests for the database which you could extend

File: src/NLog/Targets/DatabaseTarget.cs [hunk 1]
```diff
@@ -564,6 +575,32 @@ private void WriteEventToDatabase(LogEventInfo logEvent)
...
+                case DbType.Date:
+                case DbType.Time:
+                    return DateTime.ParseExact(value, GetDateFormat(par), null);
+            }
+            return value;
+        }
+        /// <summary>
+        /// Convert valut to DbType value.
+        /// </summary>
+        protected virtual string GetDateFormat(DatabaseParameterInfo par)
```
Comment (304NotModified) [inline src/NLog/Targets/DatabaseTarget.cs]: why is this one virtual?

File: src/NLog/Targets/DatabaseTarget.cs [hunk 2]
```diff
@@ -365,6 +365,15 @@ protected override void InitializeTarget()
...
                 this.SetConnectionType();
             }
+            foreach (var parameter in this.Parameters)
+            {
+                if (string.IsNullOrEmpty(parameter.DbType))
+                {
+                    parameter.DbTypeInternal = DbType.String;
+                    continue;
+                }
+                parameter.DbTypeInternal = (DbType)Enum.Parse(typeof(DbType), parameter.DbType);
```
Comment (304NotModified) [inline src/NLog/Targets/DatabaseTarget.cs]: maybe parse on setter of `parameter.DbType `?

File: src/NLog/Targets/DatabaseParameterInfo.cs [hunk 3]
```diff
@@ -79,6 +81,26 @@ public DatabaseParameterInfo(string parameterName, Layout parameterLayout)
...
+        /// </summary>
+        /// <docgen category='Parameter Options' order='10' />
+        [DefaultValue("String")]
+        public string DbType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the database parameter type.
+        /// </summary>
+        /// <docgen category='Parameter Options' order='10' />
+        internal DbType DbTypeInternal { get; set; }
```
Comment (304NotModified) [inline src/NLog/Targets/DatabaseParameterInfo.cs]: Maybe public, to set from the API (C#). 

Need another name then

**Comment by hubo0831** [comment]
确实不是很完善

**Comment by 304NotModified** [comment]
I think it's a good start!

Please post (also) google translates, thanks!

**Pull request closed without merging**